pkg/registry/resolver.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	api "github.com/openshift/ci-tools/pkg/api"
+	types "github.com/openshift/ci-tools/pkg/steps/types"
+)
+
+// Registry will hold all the registry information needed to convert between the
+// user provided configs referencing the registry and the internal, complete
+// represenation
+type Registry struct{}
```

Comment:
Using an interface here will allow us to mock out this functionality in tests that do need to validate the internals.

If we have:

```go
type Resolver interface {
    Resolve(config api.MultiStageTestConfiguration) (types.TestFlow, []error)
}

type registry struct{}

func (r *registry) Resolve(config api.MultiStageTestConfiguration) (types.TestFlow, []error) {}

func NewResolver() Resolver {
  return &registry{}
}
```

Then, a downstream user can require something like this:

```go
type SomeCIOperatorThing struct {
    resolver registry.Resolver
}
```

This lets us pass in `NewResolver()` in the production code, then mock out the functionality otherwise in tests. For instance, we might write a test that CI Operator actually contacts the resolver, so we could write a test mock like:

```go
type recordingResolver struct {
    requests []api.MultiStageTestConfiguration
    mockResponse types.TestFlow
    mockError []error
}

func (r *recordingResolver) Resolve(config api.MultiStageTestConfiguration) (types.TestFlow, []error) {
    r.requests = append(r.requests, config)
    return r.mockResponse. r.mockError
}
```

This allows us, in a test, to use dependency injection to:

 - record the calls to the resolver and assert that we see what we want
 - output a specific response and assert that the downstream logic uses it correctly
 - output an error and assert that it is handled correctly